tests/objects/test_generic_objects.py

```diff
@@ -123,6 +124,26 @@ class Connection(Generic[T]):
     assert field_copy.type.of_type._type_definition.name == "StrEdge"
 
 
+def test_list_inside_generic():
+    T = TypeVar("T")
+
+    @strawberry.type
+    class Value(Generic[T]):
+        valuation_date: datetime.date
+        value: T
+
+    @strawberry.type
+    class Foo:
+        strings: Value[List[str]]
+
+    definition = Foo._type_definition
+    assert definition.name == "Foo"
+    assert not definition.is_generic
+    [field] = definition.fields
+    assert field.python_name == "strings"
+    assert field.type._type_definition.name == "StrValue"
```

Comment:
I would expect the name to be of type `StrListValue` since the name of `List[str]` should be `StrList`, no?

What would happen if I defined my `Foo` class like that:
```python
    @strawberry.type
    class Foo:
        strings: Value[List[str]]
        string: Value[str]
```

I guess both types `strings` and `string` be called `StrValue` then. I'm not a GraphQL expert but I'd expect this to cause problems, no?

Reply:
Fixed by #1395